static/js/message_fetch.js

```diff
@@ -178,8 +178,8 @@ exports.load_messages = function (opts) {
     }
     let data = {anchor: opts.anchor, num_before: opts.num_before, num_after: opts.num_after};
 
-    if (opts.msg_list.narrowed && narrow_state.active()) {
-        let operators = narrow_state.public_operators();
+    if ((opts.msg_list.narrowed && narrow_state.active()) || opts.force_fetch) {
+        let operators = opts.msg_list.data.filter.public_operators();
         if (page_params.narrow !== undefined) {
             operators = operators.concat(page_params.narrow);
         }
```

Comment:
I think there's an opportunity to make this if/else clause much simpler and just look at `opts.msg_list.data.filter.public_operators` and `page_params.narrow`.  

(Or really, probably `page_params.narrow` should be managed somewhere else in the codebase)